src/SprykerSdk/AsyncApi/Code/Builder/AsyncApiBuilder.php

```diff
@@ -282,9 +282,11 @@ protected function addMessageToChannelType(array $asyncApi, string $messageName,
     {
         if (isset($asyncApi['channels'][$channelName][$channelType]['message'])) {
             if (isset($asyncApi['channels'][$channelName][$channelType]['message']['oneOf'])) {
-                $asyncApi['channels'][$channelName][$channelType]['message']['oneOf'][] = [
-                    '$ref' => sprintf('#/components/messages/%s', $messageName),
-                ];
+                if ($this->messageNameExists($messageName, $asyncApi['channels'][$channelName][$channelType]['message']['oneOf']) === false) {
+                    $asyncApi['channels'][$channelName][$channelType]['message']['oneOf'][] = [
+                        '$ref' => sprintf('#/components/messages/%s', $messageName),
+                    ];
+                }
```

Comment:
The level of nesting is too much now and even was before. Use early return if possible or any other strategy to have a max nesting of 2 ideally 1.

Reply:
I changed the logic of the method. It checks if `$asyncApi['channels'][$channelName][$channelType]['message']['oneOf']` does NOT exist and then if the message is in the existing array.

Reply:
I reduced the nesting from 3 levels to 1.
There are no extra checks and the conditionals go from the more abstract to the more concrete.